website/docs/language/functions/filesha1.mdx

```diff
@@ -1,19 +1,49 @@
 ---
 page_title: filesha1 - Functions - Configuration Language
-description: |-
-  The filesha1 function computes the SHA1 hash of the contents of
-  a given file and encodes it as hex.
+description: he filesha1 function computes the SHA1 hash of the contents of a given file and encodes it as hex.
 ---
 
-# `filesha1` Function
+# `filesha1` function reference
+This topic provides reference information about the `filesha1` function, which calculates the SHA-1 hash of a file's contents. 
 
-`filesha1` is a variant of [`sha1`](/terraform/language/functions/sha1)
-that hashes the contents of a given file rather than a literal string.
+## Introduction
+
+The `filesha1` is a variant of [`sha1`](/terraform/language/functions/sha1) that hashes the contents of a given file rather than a literal string. 
 
 This is similar to `sha1(file(filename))`, but
 because [`file`](/terraform/language/functions/file) accepts only UTF-8 text it cannot be used to
 create hashes for binary files.
```

Comment:
```suggestion
Use the `filesha1` function instead of wrapping the `file` function in a `sha1` function, for example `sha1(file(filename))`, because [`file`](/terraform/language/functions/file) accepts only UTF-8 text. As a result, you cannot use `sha1(file(filename))`  to create hashes for binary files.
```
This is one falls under the 'don't make assumptions' category. Making this type of comparison assumes that the reader knows (even though they probably do) about `sha1(file(filename))`.